releases-implementation/README.md

```diff
@@ -0,0 +1,415 @@
+
+##### 4.2.2.2. Scenario 1: new provider-independent app patch
+
+Now in the case above, let’s say we want to release a new patch version for some provider-independent app, and we have to do that in 5 providers, and for 2 major releases for every provider. The process would look like the following:
+- Renovate updates the version of the app in 2 git branches in the cluster chart. Here we have 2 pull requests, both created by Renovate.
+	- In case the patch is a change in the cluster chart itself (and not a new app patch version bump), we would have 1 PR to make the change in the main branch, and another PR to cherry-pick the change to the previous major version branch. Overall even more work then when “just” releasing new app patch versions.
+- We release new patch versions for the last 2 major releases of the cluster chart. Here we have 2 release pull requests, both triggered manually by pushing a new branch which then triggers the CI that opens the release pull request.
+- For all 5 providers, Renovate opens PRs for the last 2 major versions. Here we have 10 PRs opened by Renovate.
+- We release new patch versions for the last two major releases of all cluster-$provider apps. Here we have 10 release pull requests.
+
+In total, in the above scenario, we have 12 Renovate PRs and 12 release PRs, so 24 PRs in total. And all these PRs are across multiple components owned by multiple teams, meaning the some actions should be taken by people from multiple teams.
+
+The above example might seem like exaggerated. Even if you cut it in half, 12 PRs is still a lot of work.
+
+Even in one of the simplest and common scenarios, where a simple provider-independent app patch is delivered to just the latest major version of 5 providers, we would have:
+- 1 Renovate PR to bump the app version in the cluster chart,
+- 1 release PR to release the change in cluster chart,
+- 5 Renovate PRs to bump the version of the app in all cluster-$provider apps,
+- 5 release PRs to release a new patch version of all cluster-$provider apps.
+
+Here we still have to deal with 12 pull requests. Before we have introduced the cluster chart, it would be 10 PRs in default-apps-$provider repos, so 2 PRs less, but that a minor difference.
+
+Now let’s see how the above scenario would look like in a release process where releases repository is used. When a new app patch version is releases:
+- We open 1 PR in the releases repository and in that PR we add new Releases for every affected major version of every provider.
+
+That’s it - 1 PR. 1 PR for any number of releases across any number of providers.
+
+We can also maintain a draft PR in the releases repository where we automatically create next draft releases for all providers, and then Renovate can bump version numbers in those draft releases, and we just decide when to cut the new release (and name it appropriately).
```

Comment:
Ideally it should be so cheap to cut a new release that we just do it on any change and not batch changes within versions, batching will happen naturally by consecutively releasing patches, but at least we'd have atomic updates, which could be tested independently.